pkg/lib/cockpit.d.ts

```diff
@@ -121,10 +121,22 @@ declare module 'cockpit' {
         close(options?: JsonObject): void;
     }
 
-    interface ChannelOptions {
-        payload: string;
+    // these apply to all channels
+    interface BaseChannelOptions {
         superuser?: "try" | "require";
         [_: string]: JsonValue | undefined;
+        binary?: boolean,
+
+        // for remote channels
+        host?: string;
+        user?: string;
+        password?: string;
+        session?: "shared" | "private";
+    }
+
+    // this applies to opening a generic channel() with explicit payload
+    interface ChannelOptions extends BaseChannelOptions {
```

Comment:
...and I'd have renamed this one to `ChannelOpenOptions` or so.  It's a much more specific use-case.

Reply:
Done.